File: PHP/Tarjeta.php
<?php
require_once('payment.php');
class Tarjeta extends Payment{
    public $cvv ;
    public $noTarjeta ;
    public $fecha;
    
    public function __construct($id,$cvv,$noTarjeta,$fecha){
       parent::__construct($id);
        $this->cvv = $cvv;
        $this->noTarjeta = $noTarjeta;
        $this->fecha = $fecha;
     
    }
    public function ImprimirPayment(){
        echo "El metodo de pago es Tarjeta en {$this->fecha}";
      }
} ?>

File: JAVA/Tarjeta.java
package JAVA;

public class Tarjeta extends Payment{
    Integer noTarjeta;
    Integer cvv;
    String fecha;
    
    public Tarjeta(Integer id,String fecha,Integer noTarjeta,Integer cvv){
        super(id);
        this.fecha = fecha;
        this.noTarjeta = noTarjeta;
        this.cvv = cvv;

    }
    public void ImprimirPayment(){

        System.out.println("El metodo de pago es TARJETA");
    }
    
}
